detailed_docs/TRS/RC/OnRCStatus.md

```diff
@@ -0,0 +1,163 @@
+## Functional Requirements  
+
+1.
+In case
+- RC functionality is allowed on HMI  
+- all RC modules are free
+
+AND RC application sends ButtonPress or SetInteriorVehicleData request 
+
+SDL must  
+send OnRCStatus notification with RC modules in 'freeModules' and 'allocatedModules' parameters to HMI  
+send OnRCStatus notification with 'allowed = true', 'freeModules' and 'allocatedModules' parameters to mobile application 
+
+2.
+In case   
+- RC functionality is disallowed on HMI  
+- all RC modules are free
+
+AND RC application sends ButtonPress or SetInteriorVehicleData request 
+
+SDL must  
+send OnRCStatus notification with 'allowed = false' and empty arrays of 'freeModules', 'allocatedModules' to registered mobile application  
+
+3. 
+In case
+- RC functionality is allowed on HMI  
+- all RC modules are free
+
+AND non-RC application sends ButtonPress or SetInteriorVehicleData request
+
+SDL must  
+NOT send OnRCStatus to HMI and to registered non-RC application
+
+4. 
+In case  
+- RC functionality is allowed on HMI  
+- all RC modules are free
+- RC application is registered with SDL
+
+AND RC application allocates module_1  
+
+SDL must  
+send OnRCStatus to
+HMI (app_1, allocatedModules:module_1; freeModules:module_2, module_3); (app_2, allocatedModules:(); freeModules:module_2, module_3);
+
+RC app_1 (allocatedModules:module_1; freeModules:module_2, module_3);
+
+RC app_2 (allocatedModules:(); freeModules:module_2, module_3)
+
+
+5. 
+In case  
+- RC functionality is allowed on HMI 
+- RC application is registered with SDL 
+- RC application allocates module_1
+
+AND RC application unregisters or stops controlling module_1(goes to HMI level NONE)
+
+SDL must  
+send OnRCStatus to HMI and to RC application with module_1 in `freeModules`
+
+6.  
+In case  
+- RC functionality is allowed on HMI   
+- all RC modules are free
+- RC app_1 is registered with SDL
+
+AND RC app_2 starts registration
+
+SDL must  
+send OnRCStatus notification with all modules in `freeModules` parameter to HMI and to RC app_1, RC app_2 with all modules in `freeModules` parameter.  
```

Comment:
why send to app_1? there is nothing change
 shall send only to the newly registered app_2, I think HMI also not need it

Reply:
@yang1070, thank you
please, see updates https://github.com/smartdevicelink/sdl_requirements/pull/89/commits/9a23cb27fb2e940fd3601f71687d8665ce8618e6